activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -1,5 +1,7 @@
 module ActiveRecord
   class LogSubscriber < ActiveSupport::LogSubscriber
+    cattr_accessor :log_query_callers
```

Comment:
@rafaelfranca Let me know if that's what you had in mind. I did allow the LogSubscriber to define whether it logs query callers but that seemed like a tricky thing to set in something like `development.rb` so I also added an ActiveRecord::Core config that supersedes it. 

I couldn't find great examples of high level configs touching low-level configs, and while people can configure the Rails logger, it seems weird to let them reach as far down as the LogSubscriber for ActiveRecord to configure this.